src/formatter.ts

```diff
@@ -0,0 +1,648 @@
+        const traitsFormatted = contract.traits
+            .map((trait) => trait.value)
+            .join(", ");
+        this.increaseIndent();
+        const bodyFormatted = contract.declarations
+            .map((dec, index, array) => {
+                const formattedDec = this.ppContractBody(dec);
+                const nextDec = array[index + 1];
+                if (
+                    (dec.kind === "def_constant" &&
+                        nextDec?.kind === "def_constant") ||
+                    (dec.kind === "def_field" && nextDec?.kind === "def_field")
+                ) {
+                    return formattedDec;
+                }
+                return formattedDec + "\n";
+            })
+            .join("\n");
+        this.decreaseIndent();
+        const header = traitsFormatted
+            ? `contract ${contract.name} with ${traitsFormatted}`
+            : `contract ${contract.name}`;
+        const attrsRaw = contract.attributes
+            .map((attr) => `@interface("${attr.name.value}")`)
+            .join(" ");
+        const attrsFormatted = attrsRaw ? `${attrsRaw} ` : "";
+        return `${this.indent()}${attrsFormatted}${header} {\n${bodyFormatted}${this.indent()}}`;
+    }
+
+    ppContractBody(
+        declaration:
+            | ASTField
+            | ASTFunction
+            | ASTInitFunction
+            | ASTReceive
+            | ASTConstant,
+    ): string {
+        switch (declaration.kind) {
+            case "def_field":
+                return this.ppASTField(declaration);
+            case "def_function":
+                return this.ppASTFunction(declaration);
+            case "def_init_function":
+                return this.ppASTInitFunction(declaration);
+            case "def_receive":
+                return this.ppASTReceive(declaration);
+            case "def_constant":
+                return this.ppASTConstant(declaration);
+            default:
+                return `Unknown Contract Body Type: ${declaration}`;
+        }
+    }
+
+    public ppASTFunction(func: ASTFunction): string {
+        const argsFormatted = func.args
+            .map((arg) => `${arg.name}: ${this.ppASTTypeRef(arg.type)}`)
+            .join(", ");
+        const attrsRaw = func.attributes.map((attr) => attr.type).join(" ");
+        const attrsFormatted = attrsRaw ? `${attrsRaw} ` : "";
+        const returnType = func.return
+            ? `: ${this.ppASTTypeRef(func.return)}`
+            : "";
+        this.increaseIndent();
+        const stmtsFormatted = func.statements
+            ? func.statements
+                  .map((stmt) => this.ppASTStatement(stmt))
+                  .join("\n")
+            : "";
+        const body = func.statements
+            ? `{\n${stmtsFormatted}`
+            : ";";
+        this.decreaseIndent();
+        return `${this.indent()}${attrsFormatted}fun ${func.name}(${argsFormatted})${returnType} ${body}\n${this.indent()}}`;
```

Comment:
```suggestion
            const body = func.statements
      ? ` {\n${stmtsFormatted}\n${this.indent()}}`
      : ";";
    this.decreaseIndent();
    return `${this.indent()}${attrsFormatted}fun ${func.name}(${argsFormatted})${returnType}${body}`;
```

We also need a test for this, e.g.:
```
trait Q_e__dyxVu {
  abstract get fun z_(Uls9: String): String;
}
```